File: backend/routers/safety.py
import logging
from typing import Optional

from fastapi import APIRouter, Depends

from backend.auth import get_current_user
from backend.db import get_db_connection
from backend.services import safety as safety_service

logger = logging.getLogger("kabadilink.routers.safety")
router = APIRouter(tags=["Safety"])


@router.get("/safety-content")
def get_safety_content(
    material_code: str,
    content_type: Optional[str] = None,
    language: str = "en",
    limit: int = 50,
    offset: int = 0,
    current_user: dict = Depends(get_current_user),
):
    with get_db_connection() as conn:
        return safety_service.list_safety_content(
            conn, material_code, content_type, language, limit, offset
        )

File: frontend/kabadiwala-connect-main/src/lib/api/safety.ts
import { request } from "./client";

export interface SafetyContentItem {
  id: number;
  material_code: string;
  content_type: string;
  language: string;
  title: string;
  content: string;
  hazard_level?: string;
  dos?: string[];
  donts?: string[];
}

export async function getSafetyContent(params: {
  material_code: string;
  content_type?: string;
  language?: string;
  limit?: number;
  offset?: number;
}): Promise<SafetyContentItem[]> {
  return request<SafetyContentItem[]>("/safety-content", {
    method: "GET",
    params,
  });
}
